source/remediation_runbooks/scripts/enable_delivery_status_logging.py

```diff
@@ -50,24 +60,26 @@ def add_roles_to_topic(logging_role, topic_arn):
     """  
     sns = connect_to_sns()
     try:
-        sns.set_topic_attributes(TopicArn=topic_arn, AttributeName=successFeedbackRoleValue, AttributeValue=logging_role)
-        sns.set_topic_attributes(TopicArn=topic_arn, AttributeName=failureFeedbackRoleValue, AttributeValue=logging_role)
+        for endpoint in endpointTypes:
+            sns.set_topic_attributes(TopicArn=topic_arn, AttributeName=f'{endpoint}SuccessFeedbackRoleArn', AttributeValue=logging_role)
+            sns.set_topic_attributes(TopicArn=topic_arn, AttributeName=f'{endpoint}FailureFeedbackRoleArn', AttributeValue=logging_role)
```

Comment:
I agree with your choice of error handling here - the change is transactional as a whole, it either fully succeeds or we try to reset to the best of our ability